x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -442,6 +552,20 @@ public static Hasher resolve(String name) {
                 return PBKDF2_500000;
             case "pbkdf2_1000000":
                 return PBKDF2_1000000;
+            case "pbkdf2_stretch":
+                return PBKDF2_STRETCH;
+            case "pbkdf2_stretch_1000":
+                return PBKDF2_STRETCH_1000;
+            case "pbkdf2_stretch_10000":
+                return PBKDF2_STRETCH;
```

Comment:
Nit: I understand this follows an existing pattern for `pbkdf2`. But strictly speaking I think it is incorrect. We have an explicit entry of `PBKDF2_STRETCH_10000` and it should be returned here. This currently works because default cost is 10K. But the `PBKDF2_DEFAULT_COST` parameter has no inherent connection to the string of "pbkdf2_stretch_10000". That is, if someone updates `PBKDF2_DEFAULT_COST`, there is no guarantee or enforcement that this part of code would be updated accordingly as well.

Reply:
Personally, I think our fixed strings & Hasher enum has outlived its usefulness.
I would prefer we moved to something like:
`hasher:cost` or `hasher+options:cost`

So we could have
"pbkdf2:10000" but also "pbkdf2:15000" or even "pbkdf2:12345".
And then
"pbkdf2+sha512:10000"
 
But, I think that's too much change to push into this PR. I just think we should think about revisiting this in the future.

Reply:
Agreed with Tim, but also +1 to change this line as Yang suggests.